### PR TITLE
Address compat issue for collection loading on py26

### PR DESCRIPTION
**Pull request by sivel**
##### SUMMARY
Address compat issue for collection loading on py26

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
lib/ansible/plugins/loader.py

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by gundalow** [comment]
Test case `ansible-test integration  --docker centos6  -vvvv authorized_key`

File: lib/ansible/utils/import_module.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Copyright (c) 2020 Matt Martz <matt@sivel.net>
```
Comment (abadger) [inline lib/ansible/utils/import_module.py]: If we plan on keeping this long term, let's put this in `module_utils/compat/import_module.py`  If we plan on getting rid of this as soon as we can actually stop hacking the controller to work on python-2.6, let's name it something that shows it shouldn't be used by others and when it should be removed.  Maybe: `utils/_compat26.py` ?

File: lib/ansible/utils/import_module.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    from importlib import import_module
+except ImportError:
+    # importlib.import_module returns the tail
+    # whereas __import__ returns the head
+    # compat to work like importlib.import_module
+    def import_module(name):
+        module = __import__(name)
+        for part in name.split('.')[1:]:
+            module = getattr(module, part)
+        return module
```
Comment (mattclay) [inline lib/ansible/utils/import_module.py]: Is the following perhaps a better implementation?

https://github.com/ansible/ansible/blob/4e8b240b8b74db2453851f12ec82c7f782289187/test/lib/ansible_test/_data/sanity/import/importer.py#L30-L32

Comment (sivel) [reply lib/ansible/utils/import_module.py]: Hm, I think so, I didn't think about doing that

Comment (mattclay) [reply lib/ansible/utils/import_module.py]: It turns out we already have this:

https://github.com/ansible/ansible/blob/4e8b240b8b74db2453851f12ec82c7f782289187/lib/ansible/module_utils/six/__init__.py#L89-L92

Comment (mattclay) [reply lib/ansible/utils/import_module.py]: Unfortunately it is apparently private, as it is undocumented, as @sivel noticed.

File: lib/ansible/utils/import_module.py (hunk 1 above)
Comment (sivel) [reply lib/ansible/utils/import_module.py]: Done.

File: lib/ansible/module_utils/compat/importlib.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2020 Matt Martz <matt@sivel.net>
+# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)
+
+# Make coding more python3-ish
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import sys
+
+# HACK: keep Python 2.6 controller tests happy in CI until they're properly split
```
Comment (mattclay) [inline lib/ansible/module_utils/compat/importlib.py]: Now that we're making this a module_util, this should no longer be considered a hack.

Comment (mattclay) [inline lib/ansible/module_utils/compat/importlib.py]:
```suggestion
```

**Review by mattclay — APPROVED** [review]

**Pull request merged**